### PR TITLE
Fix const-correctness

### PR DESCRIPTION
**Pull request by thbeu**

### CODE REVIEW

File: shapefil.h [hunk 1]
```diff
@@ -27,22 +27,25 @@ extern "C"
...
+#define SHAPELIB_VERSION_NUMBER                                                \
+    SHAPELIB_MAKE_VERSION_NUMBER(SHAPELIB_VERSION_MAJOR,                       \
+                                 SHAPELIB_VERSION_MINOR,                       \
+                                 SHAPELIB_VERSION_MICRO)
 
-#define SHAPELIB_AT_LEAST(major, minor, micro) \
-    (SHAPELIB_VERSION_NUMBER >= SHAPELIB_MAKE_VERSION_NUMBER(major, minor, micro))
+#define SHAPELIB_AT_LEAST(major, minor, micro)                                 \
+    (SHAPELIB_VERSION_NUMBER >=                                                \
+     SHAPELIB_MAKE_VERSION_NUMBER(major, minor, micro))
```
Comment (rouault) [inline shapefil.h:48]: unrelated changes

Comment (thbeu) [reply shapefil.h:48]: Yes, someone did not run the pre-commit hooks.
